lib/puppet/util/windows/adsi.rb

```diff
@@ -97,18 +97,18 @@ def execquery(query)
       [:lpwstr, :lpdword], :win32_bool
   end
 
-  class User
-    extend Enumerable
-    extend FFI::Library
+  module Shared
+    def uri(name, host = '.')
+      if sid_uri = Puppet::Util::Windows::ADSI.sid_uri_safe(name) then return sid_uri end
 
-    attr_accessor :native_user
-    attr_reader :name, :sid
-    def initialize(name, native_user = nil)
-      @name = name
-      @native_user = native_user
+      host = '.' if ['NT AUTHORITY', 'BUILTIN', Socket.gethostname].include?(host)
+
+      account_type = self.name.split('::').last.downcase
```

Comment:
I started to introduce a new variable but moved to this instead due to some of the specs bringing the appearance that something like `Puppet::Util::Windows::ADSI::Group.uri('somegroup')` - and if they don't specify class type it was either null (or in my case I set up a default), so it could come out with an error or be wrong.